src/test/requestUser.test.js

```diff
@@ -0,0 +1,263 @@
+import {
+  collection,
+  addDoc,
+  getDocs,
+  query,
+  where,
+  updateDoc,
+  arrayUnion,
+  arrayRemove,
+  doc,
+} from "firebase/firestore";
+
+import {
+  addUserRequest,
+  acceptRequest,
+  userPendingList,
+  userAcceptedList,
+  checkUserRequest,
+  checkUserPending,
+} from "@/firebase/requestUser";
+
+const tableName = "user_request";
+
+jest.mock("firebase/firestore", () => ({
+  getFirestore: jest.fn(),
+  collection: jest.fn(),
+  addDoc: jest.fn(),
+  query: jest.fn(),
+  getDocs: jest.fn(),
+  where: jest.fn(),
+  updateDoc: jest.fn(),
+  arrayUnion: jest.fn(),
+  arrayRemove: jest.fn(),
+}));
+
+describe("User Request logic", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  [
+    {
+      user_id: "1231",
+      pending_list: ["Walter", "Joe"],
+      accepted_list: ["White"],
+    },
+    {
+      uid: "1234",
+      pending_list: ["Joe"],
+      accepted_list: ["White"],
+    },
+  ];
+
+  it("test 1: Add fromUserID to pending list of toUserID.", async () => {
```

Comment:
I think this test has the wrong name - it has the same name as the test below it (test 2), but this test *only* checks that we search up the correct user when a request is made. can the name be updated?

Reply:
test1 and test2 are positive and negative test cases both. Positive test case when an entry is found in db and negative case when the entry is not found in DB